File: java/app/src/main/java/com/ffo/mvp_java/mvp/base/BasePresenter.java
package com.ffo.mvp_java.mvp.base;


import android.util.Log;

import com.ffo.mvp_java.mvp.*;

import java.lang.ref.WeakReference;

/**
 * @author: huchunhua
 * @create_time: 2018/5/2
 * @change_time:
 * @package: com.ffo.mvp_java.mvp.imp
 * @project: java
 * @mail: huchunhua5@wasu.com
 * @describe: 一句话描述
 */
public class BasePresenter<V> implements IPresenter<V> {
    public static final String TAG = "BasePresenter";

    protected WeakReference<V> iView;

    @Override
    public void register(V view) {
        Log.i(TAG, "BasePresenter register: ");
        iView = new WeakReference<V>(view);
    }

    @Override
    public void unRegister() {
        Log.i(TAG, "BasePresenter unRegister: ");
        iView.clear();
    }
}

File: kotlin/app/src/main/java/pffo/com/mvp/imp/BasePresenter.kt
package com.ffo.ipet.mvp.imp

import com.ffo.ipet.mvp.IPresenter
import com.ffo.ipet.mvp.IView
import com.ffo.util.log.LogUtil
import java.lang.ref.WeakReference

/**
 * @author: huchunhua
 * @create_time: 2018/4/26
 * @change_time:
 * @package: com.ffo.ipet.mvp.imp
 * @project: pet
 * @mail: huchunhua5@wasu.com
 * @describe: 一句话描述
 */

open class BasePresenter<V : IView> : IPresenter<V> {
    companion object {
        val log: LogUtil = LogUtil(BasePresenter::class.simpleName.toString())
    }

    protected var iView: WeakReference<V>? = null;

    override fun register(view: V) {
        log.infoMark("------>BasePresenter bindView<------")
        iView = WeakReference<V>(view)
    }

    override fun unRegister() {
        log.infoMark("BasePresenter unBindView")
        iView?.clear()
    }
}